### ISSUE TITLE
`skore.Project` should expose ML features (e.g. `CrossValidationReporter`, etc.)

### ISSUE THREAD

**Issue by glemaitre** [probabl-ai/skore#949]
Currently, the pattern expected in `skore` is:

```python
# %%
import skore

my_project = skore.create("my_project", working_dir="/tmp")

# %%
from sklearn import datasets, linear_model
X, y = datasets.load_diabetes(return_X_y=True)
lasso = linear_model.Lasso()

# %%
from skore import CrossValidationReporter
reporter = CrossValidationReporter(lasso, X, y, cv=3)

# %%
my_project.put("reporter", reporter)
```

We should find a way that the reported is available via `my_project` directly without the need to use `.put`. It would unlock two user features:

- A user does not need to know about the `put` method if the idea is to provide a report via our UI. A user that goes with our skore-UI would probably expect that it just work out of the box. We should still allow the current pattern for advanced user that want to craft their own report.
- It is particularly relevant with the "Activity Feed": with some changes, the internal cross-validation computation could leverage a generator and the "Activity Feed" could start to report live activities as the generator is consumed. Currently the `.put` force for all results to be available to start exploring the cross-validation results. Async reporting allows to for instance break the processing if we observe that the first fold iterations go side ways. It is a gain of time. In the future, the callback API of scikit-learn would make it even easier to implement this feature.

**Comment by augustebaum** [comment]
Interesting idea; from a programming view I feel like `Project` is well scoped: it's a cache where you can `put` and `get`, and as a bonus it's somewhat aware of the kinds of objects being `put`. As such, giving it the responsibility of exposing ML features feels a bit weird to me.

**Comment by tuscland** [comment]
> We should find a way that the reported is available via my_project directly without the need to use .put. 

I agree. The vision is to store all artifacts produced by the user, irrespective of using `put` or not.

In [this discussion](https://github.com/probabl-ai/skore/discussions/577), @GaelVaroquaux suggest that items should be qualified as much as possible so that the user can make relationships, filter them, search them.

Therefore `put` is just a way to give a name to an item, that is, refine the knowledge on an item.

Also, see [my comment here](https://github.com/probabl-ai/skore/issues/963#issuecomment-2547879633) where I suggest that it would be nice to remove the need to manage a project.


> it's a cache

It is not a cache.

**Comment by MarieSacksick** [comment]
The activity feed is no more - these API changes will be discussed again once we consider the put and get again. In the mean time, there is no point in keeping this issue.
